src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -83,6 +87,10 @@ public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterC
             player.send(new RestoreFullHealthEvent(player));
             Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
             player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
+            Optional<Prefab> prefab = Assets.getPrefab("inventory");
+            StartingInventoryComponent startingInventory = prefab.get().getComponent(StartingInventoryComponent.class);
+            player.addOrSaveComponent(startingInventory);
+            player.send(new RequestInventoryEvent());
```

Comment:
Instead of adding the component and then triggering the action with the empty event I'd suggest to add the payload to the event itself, see https://github.com/Terasology/Inventory/pull/44#pullrequestreview-684342609.

This would just become 
```suggestion
            player.send(new RequestInventoryEvent(startingInventory.items));
```